### DISCUSSION TITLE
row is not more a dict and row.keys doesn’t exist in sqlalchemy 2.0

### DISCUSSION THREAD

**Discussion by mdeshmu** [sqlalchemy/sqlalchemy#9999] [Usage Questions] [answered]
I am working to make pyhive presto dialect compatible with sqlalchemy 2.0 while keeping existing compatibility with 1.3 and 1.4 intact. 

1. my code fails here when i try to access row as dict like row[“column”]
https://github.com/dropbox/PyHive/blob/master/pyhive/sqlalchemy_presto.py#L181

2. row.keys() method is also not working here.
https://github.com/dropbox/PyHive/blob/master/pyhive/tests/test_sqlalchemy_hive.py#L68

I tried row._asdict() instead of row, it works with 1.4 and 2.0 for both cases but doesn’t work with 1.3.

Is there a recommended way that can work with all three versions?

**Comment by zzzeek** [comment, marked as answer]
for 1.4 and 2.0 you use `row._mapping` or `result.mappings()` as detailed at https://docs.sqlalchemy.org/en/20/changelog/migration_14.html#change-4710-core.

this wont work in 1.3 so you would need to make a compatibility function on your end.
